File: Dev/unitTest_Engine_cs/Graphics/PostEffect/PostEffectGrayScale.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics.PostEffect
{
	class PostEffectGrayScale : EngineTest
	{
		public PostEffectGrayScale()
			: base(10)
		{
		}

		protected override void OnStart()
		{
			var scene = new asd.Scene();
			var layer = new asd.Layer2D();
			var obj = new asd.TextureObject2D();
			scene.AddLayer(layer);
			layer.AddObject(obj);
			asd.Engine.ChangeScene(scene);

			var g = asd.Engine.Graphics;
			var texture = g.CreateTexture2D("Data/Texture/Sample1.png");
			obj.Texture = texture;
			obj.Scale = new asd.Vector2DF(1, 1);

			var pe = new asd.PostEffectGrayScale();
			layer.AddPostEffect(pe);
		}
	}
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/PostEffect/PostEffectGrayScale.cpp
#include <ace.h>
#include <gtest/gtest.h>
#include <memory>
#include "../../EngineTest.h"


class Graphics_PostEffectGrayScale : public EngineTest
{
public:
	float intensity;
	Graphics_PostEffectGrayScale(bool isOpenGLMode) :
		EngineTest(ace::ToAString("PostEffectGrayScale"), isOpenGLMode, 10)
	{}
protected:
	void OnStart() override
	{
		auto scene = std::make_shared<ace::Scene>();
		auto layer = std::make_shared<ace::Layer2D>();
		auto obj = std::make_shared<ace::TextureObject2D>();
		scene->AddLayer(layer);
		layer->AddObject(obj);
		ace::Engine::ChangeScene(scene);

		auto g = ace::Engine::GetGraphics();
		auto texture = g->CreateTexture2D(ace::ToAString("Data/Texture/Sample1.png").c_str());
		obj->SetTexture(texture);
		obj->SetScale(ace::Vector2DF(2, 2));

		auto pe = std::make_shared<ace::PostEffectGrayScale>();
		layer->AddPostEffect(pe);
	}
};

ENGINE_TEST(Graphics, PostEffectGrayScale)


